test(frontend): the middleware auth test depends on AUTH_SECRET being absent instead of stubbing it

## 증상

`frontend/src/__tests__/middleware.test.ts > middleware > passes through with valid auth cookie`
가 환경에 따라 실패한다.

```
AssertionError: expected { type: 'redirect', …(1) } to deeply equal { type: 'next' }
- Expected            + Received
  { "type": "next" }    { "type": "redirect", "url": "http://localhost:3000/login" }
 ❯ src/__tests__/middleware.test.ts:97:20
```

실측 (2026-08-21, MBP):

| 실행 방법 | 결과 |
|---|---|
| `npm run test` 직접 | 5회 전부 통과 (1455/1455) |
| 래퍼 프로세스 경유 (`gstack-evidence run`) | 3회 전부 실패 (1 failed / 1454 passed) |

무작위가 아니라 **실행 환경에 갈린다.**

## 원인

`middleware.test.ts:79-88` 이 기대 토큰을 이렇게 만든다:

```ts
// AUTH_SECRET is unset so the helper falls back to DASHBOARD_PASSWORD as the HMAC key.
const expectedToken = createHmac("sha256", "secret123")
  .update("nuri-auth-token:v1")
  .digest("hex");
```

주석이 전제를 명시하고 있다 — **`AUTH_SECRET` 이 없어야** 헬퍼가 `DASHBOARD_PASSWORD`
로 폴백한다. 그런데 이 테스트는 `DASHBOARD_PASSWORD` 만 `vi.stubEnv` 하고
**`AUTH_SECRET` 은 스텁하지 않는다.** 즉 "부재" 를 강제하지 않고 **주변 환경에 맡긴다.**

같은 레포의 `auth-token.test.ts` 는 이미 올바르게 하고 있다:

```
auth-token.test.ts:48   vi.stubEnv("AUTH_SECRET", "");   // falls back to DASHBOARD_PASSWORD
auth-token.test.ts:17   vi.stubEnv("AUTH_SECRET", "shared-secret");
auth-token.test.ts:33   vi.stubEnv("AUTH_SECRET", "primary-key");
```

`AUTH_SECRET` 이 무엇이든 값을 갖게 되면(셸 export · 스텁 누수 · 워커 재사용) 헬퍼가
그걸 HMAC 키로 쓰고, `secret123` 으로 계산한 토큰은 불일치 → 미들웨어가 `/login` 으로
리다이렉트한다. 그게 관측된 실패다.

(이 머신의 대화형 셸과 로그인 셸 둘 다 `AUTH_SECRET` 은 unset 이었다. 그래서 남는
후보는 워커 간 스텁 누수/재사용 쪽이며, 어느 쪽이든 **테스트가 부재를 강제하지 않는
것**이 결함이다.)

## 왜 CI 는 초록인가

CI 러너 환경에 `AUTH_SECRET` 이 없어 우연히 전제가 성립한다. 즉 이 테스트는
**환경이 맞아떨어질 때만** 통과하는 형태이고, 통과가 코드의 정당성을 증명하지 않는다.
`.claude/rules/enforcement.md` 가 다루는 "green dead gate" 와 같은 계열 —
앰비언트 상태가 분기를 대신 밟는 문제다(`feedback_local_coverage_overstates` 참조).

## 고칠 것

`middleware.test.ts` 의 해당 테스트(및 같은 전제를 쓰는 이웃 케이스)에
`vi.stubEnv("AUTH_SECRET", "")` 을 추가해 전제를 **명시**한다. `auth-token.test.ts:48`
과 같은 형태다.

확인 방법: `AUTH_SECRET=anything npm run test` 가 수정 전에는 실패하고 수정 후에는
통과해야 한다 — 그게 이 fix 의 회귀 테스트다.

## 발견 경위

#1130/#1131 PR 의 `/ship` Step 5 에서 프론트 레인이 실패. 해당 브랜치는 프론트엔드
파일을 **하나도** 건드리지 않으므로 in-branch 실패가 아니다.

곁가지: 처음엔 `npm run test 2>&1 | tail -8` 로 감싸 돌려서 evidence ledger 가
`exit=0` 을 기록했다 — 파이프가 종료코드를 삼켰다. `bash -o pipefail -c` 로 다시
돌려서야 `exit=1` 이 잡혔다. 스크립트에서 테스트를 파이프로 감쌀 때 반복되는 함정이다.

--- a/frontend/src/__tests__/middleware.test.ts
+++ b/frontend/src/__tests__/middleware.test.ts
@@ -15,6 +15,7 @@ describe("middleware", () => {
   beforeEach(() => {
     vi.resetModules();
     vi.unstubAllGlobals();
+    vi.unstubAllEnvs();
   });
 
   it("passes through when no DASHBOARD_PASSWORD set", async () => {
@@ -75,14 +76,35 @@ describe("middleware", () => {
 
   it("passes through with valid auth cookie", async () => {
     vi.stubEnv("DASHBOARD_PASSWORD", "secret123");
+    // #1136: 앰비언트 환경에 기대지 않는다 — AUTH_SECRET 을 명시 stub 해야
+    // 러너 환경에 AUTH_SECRET 이 있어도(래퍼 프로세스 등) 결과가 갈리지 않는다.
+    vi.stubEnv("AUTH_SECRET", "test-auth-secret");
     const { middleware } = await import("@/middleware");
 
     // Generate the expected token using the same HMAC scheme as the
-    // production auth-token helper. AUTH_SECRET is unset so the helper
-    // falls back to DASHBOARD_PASSWORD as the HMAC key. The body of the
-    // hash is the static label "nuri-auth-token:v1" — password content
-    // is intentionally NOT mixed in (CodeQL would flag that as an
-    // insecure password hash regardless of the HMAC key).
+    // production auth-token helper. The body of the hash is the static
+    // label "nuri-auth-token:v1" — password content is intentionally NOT
+    // mixed in (CodeQL would flag that as an insecure password hash
+    // regardless of the HMAC key).
+    const expectedToken = createHmac("sha256", "test-auth-secret")
+      .update("nuri-auth-token:v1")
+      .digest("hex");
+
+    const request = {
+      nextUrl: { pathname: "/dashboard" },
+      cookies: { get: (name: string) => name === "nuri-auth" ? { value: expectedToken } : undefined },
+      url: "http://localhost:3000/dashboard",
+    };
+
+    const result = await middleware(request as unknown as NextRequest);
+    expect(result).toEqual({ type: "next" });
+  });
+
+  it("falls back to DASHBOARD_PASSWORD as HMAC key when AUTH_SECRET is empty", async () => {
+    vi.stubEnv("DASHBOARD_PASSWORD", "secret123");
+    vi.stubEnv("AUTH_SECRET", "");
+    const { middleware } = await import("@/middleware");
+
     const expectedToken = createHmac("sha256", "secret123")
       .update("nuri-auth-token:v1")
       .digest("hex");

